src/dag/write.ts

```diff
@@ -69,8 +70,11 @@ export class Write {
   }
 
   async commit(): Promise<void> {
-    await this.collectGarbage();
-    await this._kvw.commit();
+    if (!this._committed) {
```

Comment:
Not super happy about this.

But, db has no `withWrite`. The way it is structured is a larger refactor to make it use the with pattern.

Reply:
Maybe it should throw if you commit twice? Kinda weird to do that, probably a mistake.

Reply:
That is what I had but that doesn't work unless we refactor `db.Write` too